### ISSUE TITLE
Reopening job with Union ends in NPE when trying to configure component

### ISSUE THREAD

**Issue by LosD** [datacleaner/DataCleaner#1064]
A job with a union component (in my case the same as in #1056) saved and reloaded will throw an NPE when opening the configuration dialog:

```
Unexpected error!
 
Message:
    java.lang.NullPointerException
Level:
    SEVERE
Stack Trace:
null
    org.datacleaner.panels.fuse.StreamColumnMatrixMultipleCoalesceUnitPropertyWidget.refresh(StreamColumnMatrixMultipleCoalesceUnitPropertyWidget.java:127)
    org.datacleaner.panels.fuse.StreamColumnMatrixMultipleCoalesceUnitPropertyWidget.<init>(StreamColumnMatrixMultipleCoalesceUnitPropertyWidget.java:94)
    org.datacleaner.panels.fuse.FuseStreamsComponentBuilderPresenter.<init>(FuseStreamsComponentBuilderPresenter.java:57)
    org.datacleaner.panels.fuse.FuseStreamsComponentBuilderPresenterRenderer.render(FuseStreamsComponentBuilderPresenterRenderer.java:67)
    org.datacleaner.panels.fuse.FuseStreamsComponentBuilderPresenterRenderer.render(FuseStreamsComponentBuilderPresenterRenderer.java:41)
    org.datacleaner.windows.ComponentConfigurationDialog.getDialogContent(ComponentConfigurationDialog.java:176)
    org.datacleaner.windows.AbstractDialog.getWindowContent(AbstractDialog.java:174)
    org.datacleaner.windows.AbstractDialog.initialize(AbstractDialog.java:124)
    org.datacleaner.windows.AbstractDialog.setVisible(AbstractDialog.java:147)
    org.datacleaner.windows.AbstractDialog.open(AbstractDialog.java:97)
    org.datacleaner.widgets.visualization.JobGraphActions.showConfigurationDialog(JobGraphActions.java:81)
    org.datacleaner.widgets.visualization.JobGraphMouseListener$1.actionPerformed(JobGraphMouseListener.java:159)
    javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
    javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2346)
    javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
    javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
    javax.swing.AbstractButton.doClick(AbstractButton.java:376)
    javax.swing.plaf.basic.BasicMenuItemUI.doClick(BasicMenuItemUI.java:833)
    javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(BasicMenuItemUI.java:877)
    java.awt.Component.processMouseEvent(Component.java:6525)
    javax.swing.JComponent.processMouseEvent(JComponent.java:3321)
    java.awt.Component.processEvent(Component.java:6290)
    java.awt.Container.processEvent(Container.java:2234)
    java.awt.Component.dispatchEventImpl(Component.java:4881)
    java.awt.Container.dispatchEventImpl(Container.java:2292)
    java.awt.Component.dispatchEvent(Component.java:4703)
    java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4898)
    java.awt.LightweightDispatcher.processMouseEvent(Container.java:4533)
    java.awt.LightweightDispatcher.dispatchEvent(Container.java:4462)
    java.awt.Container.dispatchEventImpl(Container.java:2278)
    java.awt.Window.dispatchEventImpl(Window.java:2739)
    java.awt.Component.dispatchEvent(Component.java:4703)
    java.awt.EventQueue.dispatchEventImpl(EventQueue.java:746)
    java.awt.EventQueue.access$400(EventQueue.java:97)
    java.awt.EventQueue$3.run(EventQueue.java:697)
    java.awt.EventQueue$3.run(EventQueue.java:691)
    java.security.AccessController.doPrivileged(Native Method)
    java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
    java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:86)
    java.awt.EventQueue$4.run(EventQueue.java:719)
    java.awt.EventQueue$4.run(EventQueue.java:717)
    java.security.AccessController.doPrivileged(Native Method)
    java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
    java.awt.EventQueue.dispatchEvent(EventQueue.java:716)
    java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
    java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
    java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
    java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
    java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
    java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

This happens both in master and 4.5.4. The job still executes properly.

**Comment by LosD** [comment]
Just tested to make sure: This also happens if the bug in #1056 is not triggered (all columns has different names)

**Comment by LosD** [comment]
This is a complication of the fix in 1eeed14a8c1f19b8d2ffe43fe1da093f828ddd98 for unions getting the weird layout. I believe that if we revert that fix, the original issue (columns getting a weird layout when multiple columns has the same name) will be fixed along with #1056, as it seems to be the same root cause.
